doc/tutorials/trustyACRN.rst

```diff
@@ -0,0 +1,408 @@
+persistent storage (secure storage), wireless display ("cast") of
+protected content, secure PIN and fingerprint processing, and even
+malware detection.
+
+In embedded products such as an automotive IVI system, the most important
+security services requested by customers are keystore and secure
+storage. In this article we will focus on these two services.
+
+Keystore
+========
+
+Keystore (or Keymaster app in Trusty OS) provides the following
+services:
+
+- Key generation
+- Import and export of asymmetric keys (no key wrapping)
+- Import of raw symmetric keys (no key wrapping)
+- Asymmetric encryption and decryption with appropriate padding modes
+- Asymmetric signing and verification with digesting and appropriate
+  padding modes
+- Symmetric encryption and decryption in appropriate modes, including
+  an AEAD mode
+- Generation and verification of symmetric message authentication codes
+
+Protocol elements, such as purpose, mode and padding, as well as access
+control constraints, are specified when keys are generated or imported
+and are permanently bound to the key, ensuring the key cannot be used in
+any other way.
+
+In addition to the list above, there is one more service that Keymaster
+implementations provide, but which is not exposed as an API: Random
+number generation. This is used internally for generation of keys,
+Initialization Vectors (IVs), random padding, and other elements of
+secure protocols that require randomness.
+
+Using Android as an example, Keystore functions are explained in greater
+details in this `Android keymaster functions document
+<https://source.android.com/security/keystore/implementer-ref>`_
+
+.. figure:: images/trustyacrn-image3.png
+   :align: center
+   :width: 600px
+   :name: keymaster-app
+
+
+As shown in :numref:`keymaster-app` above, the Keymaster HAL is a
+dynamically-loadable library used by the Keystore service to provide
+hardware-backed cryptographic services. To keep things secure, HAL
+implementations don't perform any security sensitive
+operations/algorithms in user space, or even in kernel space. Sensitive
+operations are delegated to a secure world TEE (Trusty OS) reached
+through a kernel interface. The purpose of the Keymaster HAL is only to
+marshal and unmarshal requests to the secure world.
+
+Secure Storage (SS)
+===================
+
+Trusty implements a secure storage services (in Secure Storage TA) based
+on RPMB (Replay Protected Memory Block) partition in eMMC or UFS flash
+storage. The details of how RPMB works are out of scope in this article.
+You can read the `eMMC/UFS JEDEC specification
+<https://www.jedec.org/standards-documents/focus/flash/universal-flash-storage-ufs>`_
+to understand that.
+
+This secure storage can provide data confidentiality, integrity, and
+anti-replay protection.Confidentiality is guaranteed by data encryption
+with a root key derived from the platform chipset's unique key/secret.
```

Comment:
root key -> encryption key, to align with line 185: 
"encrypted by hardware-backed unique encryption key"